### DISCUSSION TITLE
Odometer information on 'charges' table

### DISCUSSION THREAD

**Discussion by Flo-63** [tillsteinbach/VWsFriend#341] [Ideas]
Hi,

it would be a real help to see the Odometer data in charges. Is there any way to get the data in?  This would save me from taking a photo while / before charging my ID4...

Cheers,

Flo

**Comment by tillsteinbach** [comment]
There is a way, you need to get SW 3.0 on your car. Once that is done you should immediately see the odometer. Unfortunately I have no idea how to motivate VW to push out the update :)

**Reply by Flo-63** [reply]
Great... Thanks for quicke reply  - and another reason to look forward to 3.0 - if this update ever will happen...

**Reply by MattiJarvinen** [reply]
My bet on the 3.0 OTA is that the 3-year guarantee runs out before I get that update.

**Reply by tillsteinbach** [reply]
Yes :( I am considering not buying VW anymore…

**Reply by turkano** [reply]
It appears that with 3.0, the value of the odometer is not transmitted through the API either, in HA that field appears empty.
In addition, I'm not sure if it's just me, but it does nothing with the received location, I don't know if it's because the car only sends it when it's parked and does not send a route as such. Do you know anything about this? In Tronity or in the We Connect ID app, the location does appear to me.

**Reply by tillsteinbach** [reply]
I have reports that at least with 3.2 odometer is transmitted. The location is only transmitted when the ignition is turned off. So no possibility to record routes

**Reply by turkano** [reply]
Sure, let's wait another year for the arrival of 3.2 hahaha. Yes, of course, the trips can't be recorded, but I could record the starting and ending position to store places of charge and visited places, right?

**Reply by tillsteinbach** [reply]
Yes, that should work since 3.0 with VWsFriend

**Reply by turkano** [reply]
Indeed, after restarting docker it now shows the locations. Thank you.

**Reply by tillsteinbach** [reply]
I should put that in the readme :)

**Reply by turkano** [reply]
I just discovered that if you activate “Automatic service scheduling” in We Connect ID. App it does send the odometer. 3.0 OTA Software Version 362, I haven't installed the second part yet (363)
![IMG_2695](https://user-images.githubusercontent.com/90693784/211406219-d0cb3465-368d-4bda-adf2-27dc0924606f.jpeg)
![IMG_2696](https://user-images.githubusercontent.com/90693784/211406255-51f60d8a-abaf-458f-aa38-66d2d23f5b83.jpeg)

**Reply by tillsteinbach** [reply]
I should put that in the readme!
